### DISCUSSION TITLE
关于milvus文档里没提到的几点疑问

### DISCUSSION THREAD

**Discussion by xyzkk3** [milvus-io/milvus#16148] [Q&A and General discussion]
1 如果内存已经被milvus占满了的情况下，我加载一个patrition到内存中会成功吗？会自动释放掉最早加载的partition？
2 没有事先加载collection,直接去搜索，实际体验是第一次搜索慢，后面几次快，是第一次搜索时自动加载了？
3 当内存不够整个collection加载进去时，那会加载collection部分数据吗？其余的走从磁盘搜索逻辑？

**Comment by yhmo** [comment]
哪个版本的？2.0版本要求搜索之前一定要调用一下load

**Comment by yhmo** [comment]
如果是1.x版本：
1 . cache被占满的情况下，加载另一个partition，会自动释放较早加载的partition
2. 如果没有调用load就直接搜索，在第一次搜索时会自动加载，所以第一次搜索很慢
3. 当cache不够整个collection时，加载该collection，仅能加载一部分数据，当搜索的时候，就不得不从磁盘边加载边释放数据，搜索会很慢

**Reply by xyzkk3** [reply]
1.x版本，谢谢，明白了

**Comment by susugo** [comment]
2.x版本在这块是怎么样的呢，我遇到了一次collection加载后后面掉线问题，这个是我们特殊设置的内存淘汰机制么。

**Reply by yhmo** [reply]
2.x不做缓存的swap，但可以使用mmap实现相似目的  https://milvus.io/docs/mmap.md#Use-mmap
不开mmap的话，如果内存不够load collection，它会报错(客户端load_collection()返回错误信息，或者客户端抛异常)，但server端一般不会挂掉。
